### DISCUSSION TITLE
Not able to perform testing using huff in WSL on foundry

### DISCUSSION THREAD

**Discussion by parwejkhan1816** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#8] [General]
Error: 
```
Command :  forge test --match-path *Huff* --debug testReadValue 
[⠑] Compiling...
No files changed, compilation skipped

Ran 1 test for test/V1/HorseStoreHuff.t.sol:HorseStoreHuff
[FAIL. Reason: setup failed: revert: Invalid huffc binary. Run `curl -L get.huff.sh | bash` and `huffup` to fix.] setUp() (gas: 0)
Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 2.50s (0.00ns CPU time)

Ran 1 test suite in 3.38s (2.50s CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)
Error: 
debug arena is empty
```
Steps i have taken to correct this  : 

==> I reinstall the huffc then check
==> reinstall lib extension
==> check everything in code thrice
==> update my WSL
Can somebody help ?.

**Comment by Equious** [comment]
Do you have npm or yarn installed by any chance? I believe there's an issue in binary_check.sh that throws this if those two are both absent.

https://github.com/huff-language/foundry-huff/pull/60

**Reply by parwejkhan1816** [reply]
Node is installed on windows but not in WSL

**Reply by Equious** [reply]
Take a look at the link I sent, I believe it'll lead you to some changes to binary_check.sh that'll fix things.

**Comment by parwejkhan1816** [comment]
i follow the link and do the required changes on binary_check.sh but issue still persist 
https://github.com/huff-language/foundry-huff/pull/60
![ffi](https://github.com/Cyfrin/assembly-evm-opcodes-and-formal-verification-course/assets/85105093/78bd1702-87d2-45f3-ada1-b2c9fef25201)

**Comment by MahithChigurupati** [comment]
I am facing the same issue now. were you able to solve it by any chance @parwejkhan1816 ?

**Reply by parwejkhan1816** [reply]
the issue actually occurred because of Node.js. We either need to install Node.js in our WSL, or if we have already installed it on Windows, we can use Git Bash. Git Bash is also working, but you need to install Foundry again on your Git Bash. Hope this works

**Reply by MahithChigurupati** [reply]
I am actually using mac and facing the issue and I have node, yarn, foundry, git - everything installed but still facing the issue.

**Reply by EngrPips** [reply]
Honestly, I don't know how to help here, but let's look forward to any help @PatrickAlphaC can give when they are available.

**Reply by PatrickAlphaC** [reply]
Interesting. What do you get when you run `huffc /path/to/your/huff.huff -b`?

Do you get a binary output? If you can't run the `huffc` command, the foundry tool won't work either.

**Reply by MahithChigurupati** [reply]
I can get the binary output @PatrickAlphaC, just facing issue running tests for huff

**Reply by PatrickAlphaC** [reply]
can you add `-vvvv` to your `forge test` command? That'll give a more descriptive output.
